data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/plugins/accumulator/InMemoryBuffer.java

```diff
@@ -61,4 +62,14 @@ public void writeEvent(byte[] bytes) throws IOException {
         byteArrayOutputStream.write(System.lineSeparator().getBytes());
         eventCount++;
```

Comment:
@mallikagogoi7 , If we are going to require setting the event count, then this line should be removed. Also, other users of this interface may be expecting this behavior. We need to be sure this is consistently working for all sinks/codecs.

Reply:
@dlvenable there are 3 threshold properties (event_count,event_collect_timeout and maxmum_size) that user can configure in pipeline yaml.
So All 3 properties will accumulate the values based on the incoming record events, which ever property will met the threshold limit first, The accumulated events will flueshed to the respective endpoint. 

Here we are incrementing eventCount++ to meet the threshold limit based on user configuration in the yaml. 

This threshold behaviour same as s3 sink threshold  behaviour.

Reply:
@dlvenable InMemoryBuffer writeEvent() implemention is replaced with codec writeEvent() implementations. This implemenIntation is currently not being used in any plugin.  So, shall I remove this method, as its not used?

Reply:
@mallikagogoi7 , If we are not using this, then let's delete it.